pkg/services/ngalert/models/alertmanager.go

```diff
@@ -20,11 +21,18 @@ type GetLatestAlertmanagerConfigurationQuery struct {
 	Result *AlertConfiguration
 }
 
+// GetAllValidAlertmanagerConfigurationsQuery is the query to get all valid configurations for a given organization.
+type GetAllValidAlertmanagerConfigurationsQuery struct {
+	OrgID  int64
+	Result []*AlertConfiguration
+}
+
 // SaveAlertmanagerConfigurationCmd is the command to save an alertmanager configuration.
 type SaveAlertmanagerConfigurationCmd struct {
 	AlertmanagerConfiguration string
 	FetchedConfigurationHash  string
 	ConfigurationVersion      string
 	Default                   bool
 	OrgID                     int64
+	IsValid                   bool
```

Comment:
The tricky part is that what's considered valid might drift over time.. a valid config in one release might not be valid a few releases down the line as we add migrations.

Honestly, I'm not sure if there's a great solution to that problem though, especially since we evict these configs once they get old enough.

I'm not sure if I have any specific feedback here, other than we know the problem exists, I think this PR is definitely the right direction regardless, just thinking out loud a bit 🤷 

Maybe we rename this column to something like `loaded` or `successfully_loaded` to capture this a little better, it may not be valid but we did know it worked at one point in the past. WDYT?

Reply:
That's a great idea! I like the name `successfully_loaded`, I think it's the most accurate one for this column.